test-cypress/integration/explore/visual.graph.spec.js

```diff
@@ -1,774 +1,760 @@
+						cy.get('g[class="node-wrapper"]')
+							.its('length')
+							.should('to.be', '2');
+					});
+			});
+	});
+
+	it('Test remove parent node', function () {
+		// Search for an valid resource
+		searchForResource(validResource);
+		// Verify redirection to existing visual graph
+		cy.url({timeout: 5000})
+			.should('include', validResource);
+		// Verify that search bar isn't visible
+		cy.get('div[class="card mb-2"] input')
+			.should('not.be.visible');
+		// Hover over node with the mouse
+		cy.get('g[id="http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#' + validResource + '"] circle', {waitForAnimations: true, animationDistanceThreshold: 0})
+			.should('be.visible', {timeout: 5000}).wait(5000)
+			.trigger('mouseover').wait(1000)
+			.then(() => {
+				// Select remove function
+				cy.get('g.nodes-container > g.menu-events > g.close-icon > circle')
+					.trigger('click', {force: true});
+				// Verify that the search bar re-appears on the screen
+				cy.get('div[class="row slider mb-2"] input')
+					.should('be.visible');
+			});
+	});
+
+	it('Test expand collapsed node which has connections with double click', function () {
+
+		// Search for an valid resource
+		searchForResource(validResource);
+		// Verify redirection to existing visual graph
+		cy.url({timeout: 5000}).should('include', validResource);
+
+		// Hover over the node with the mouse
+		cy.get('g[id="http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#' + validResource + '"] circle', {waitForAnimations: true, animationDistanceThreshold: 0})
+			// timeout is needed because mouseover event will result in
+			// pop-up of menu icons only if nodes are not moving
+			.should('be.visible', {timeout: 5000}).wait(5000)
+			.trigger('mouseover').wait(1000);
+
+		// Select collapse function
+		cy.get('g.nodes-container > g.menu-events > g.collapse-icon > circle')
+			.trigger('click', {force: true});
+		// Verify that all links to the USRegion node are collapsed
+		cy.get('text[class="predicate"]')
+			.should('have.length', '0');
+		// Verify that the USRegion node is the only node left in the graph
+		cy.get('g[class="node-wrapper"]')
+			.should('have.length', '1').and('contain', 'USRegion');
+
+		// Double click on collapsed node
+		cy.get('g[id="http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#' + validResource + '"] circle')
+			.click({force: true})
+			.then(() => {
+				cy.get('g[id="http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#' + validResource + '"] circle')
+					.click({force: true});
+			});
+
+		// Verify that all links to the USRegion node are expanded
+		cy.get('text[class="predicate"]')
+			.should('have.length', '2');
+		// Verify that the USRegion node is not the only node left in the graph
+		cy.get('g[class="node-wrapper"]')
+			.its('length')
+			.should('be.gt', '1');
+	});
+
+	it('Test verify mouse/keyboard actions', function () {
+		let mouseActions = 'Mouse actions\n                ' +
```

Comment:
These probably can be extracted in text files. Or just have the actual text normalized before the assertion, otherwise its just insane to have and maintain these. Also I'm almost 100% sure that this formatting wouldn't be the same under different browsers.